default/generated/spring-framework/spring-framework-5.x-to-6.0-data-access.yaml

```diff
@@ -29,6 +29,71 @@
   - title: 'Spring 6.0 migration guide'
     url: https://github.com/spring-projects/spring-framework/wiki/Spring-Framework-6.0-Release-Notes#removed-apis
 
+- ruleID: spring-framework-5.x-to-6.0-data-access-00002
+  category: mandatory
+  effort: 1
+  labels:
+  - konveyor.io/source=spring5
+  - konveyor.io/target=spring6+
+  - konveyor.io/target=hibernate6+
+  when:
+    or:
+    - builtin.filecontent:
+        filePattern: (hibernate\.properties|persistence\.xml|cfg\.xml|application\.properties|application\.yaml)
+        pattern: hibernate.bytecode.provider=javassist
```

Comment:
I think this pattern would only be valid for `*.properties` files, right?

Reply:
I believe you can set this in xml files as well (persistence.xml as well as I believe you can use application.xml instead of application.properties).  I know it isn't used all that often but figured why not.  Let me know if you want me to just change it to `*.properties`

Reply:
I'm at least sure that the pattern `hibernate.bytecode.provider=javassist` won't be detected in `*.yaml` files, since it would look more like
```yaml
hibernate:
  bytecode:
    provider: javassist
```
or simply `hibernate.bytecode.provider: javassist` with `:` instead of `=`. Not sure about the `persistence.xml` file, but it would probably look more like `<property name="hibernate.bytecode.provider">javassist</property>` or something similar maybe?

Reply:
True.  So the question is should I remove the xml and yaml portion or create another pattern?

Reply:
After thinking about it I don't believe that property would be in the persistence.xml file.  That's primarily JPA specific configs not hibernate.  So we could probably just change the rule to this:
```
- builtin.filecontent:
    filePattern: *.properties
    pattern: hibernate.bytecode.provider=javassist
- builtin.filecontent:
    filePattern: application\.yaml
    pattern: hibernate.bytecode.provider: javassist
 ```
 If your good with that I'll make the change or if you just think *.properties is good enough we could remove the application.yaml check.  Let me know